app/src/main/java/com/example/filemanager/ListAdapter.kt

```diff
@@ -20,14 +29,16 @@ internal class ListAdapter : RecyclerView.Adapter<ListAdapter.ViewHolder>() {
     }
 
     override fun onBindViewHolder(holder: ViewHolder, position: Int) {
-        val item = items[position]
-        holder.name.text = item.fileName
+        val item = list[position]
+        holder.icon.setImageDrawable(item.icon?.let { ContextCompat.getDrawable(context, it) })
```

Comment:
Is it possible for the icon to be null? Can it be converted to non nullable type?

Reply:
I converted the icon to non nullable type